File: leetcode/dp/knapsack/0494_target_sum.py
# LC 494. Target Sum | Medium (Knapsack variant)
class Solution:
    def findTargetSumWays(self, nums, target):
        total = sum(nums)
        if (total + target) % 2 or abs(target) > total: return 0
        s = (total + target) // 2; dp = [0]*(s+1); dp[0] = 1
        for n in nums:
            for j in range(s, n-1, -1): dp[j] += dp[j-n]
        return dp[s]

File: leetcode/dp/knapsack/0494_target_sum.cpp
// LC 494. Target Sum | Medium (Knapsack variant)
#include <vector>
using namespace std;

class Solution {
public:
    int findTargetSumWays(vector<int>& nums, int target) {
        int total = 0; for (int n : nums) total += n;
        if ((total+target)%2 || abs(target) > total) return 0;
        int s = (total+target)/2;
        vector<int> dp(s+1, 0); dp[0] = 1;
        for (int n : nums) for (int j = s; j >= n; j--) dp[j] += dp[j-n];
        return dp[s];
    }
};
